src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,113 @@
-export const PeopleFilters = () => {
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { useCallback, useEffect } from 'react';
+import { SearchLink } from './SearchLink';
+import { getSearchWith } from '../utils/searchHelper';
+import { Person } from '../types';
+
+type Props = {
+  people: Person[],
+  setFilterPeople: (people: Person[]) => void,
+};
+
+const startedCenturies = ['16', '17', '18', '19', '20'];
+
+export const PeopleFilters: React.FC<Props> = ({
+  people,
+  setFilterPeople,
+}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+
+  const handlerCenturiesChange = useCallback((el: string) => {
+    if (centuries.includes(el)) {
+      return { centuries: [...centuries].filter(cent => cent !== el) || [] };
+    }
+
+    return { centuries: [...centuries, el] };
+  }, [centuries]);
+
+  const handlerQuery = useCallback((
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const newParams = { query: event.target.value || null };
+
+    setSearchParams(getSearchWith(searchParams, newParams));
+  }, [searchParams]);
+
+  const getFilteredPeople = useCallback(() => {
```

Comment:
you  shouldn't do it in this component
you can get `searchParams` from any component and filter array to render it